pkg/controller/logstorage/eskubecontrollers.go

```diff
@@ -0,0 +1,136 @@
+	install *operatorv1.InstallationSpec,
+	hdler utils.ComponentHandler,
+	reqLogger logr.Logger,
+	managementClusterConnection *operatorv1.ManagementClusterConnection,
+	managementCluster *operatorv1.ManagementCluster,
+	authentication *operatorv1.Authentication,
+	esLicenseType render.ElasticsearchLicenseType,
+	ctx context.Context,
+) (reconcile.Result, bool, error) {
+	kubeControllerEsPublicCertSecret, err := utils.GetSecret(ctx, r.client, relasticsearch.PublicCertSecret, rmeta.OperatorNamespace())
+	if err != nil {
+		log.Error(err, err.Error())
+		r.status.SetDegraded("Failed to get Elasticsearch pub cert secret used by kube controllers", err.Error())
+		return reconcile.Result{}, false, err
+	}
+
+	kubeControllersUserSecret, err := utils.GetSecret(ctx, r.client, kubecontrollers.ElasticsearchKubeControllersUserSecret, rmeta.OperatorNamespace())
+	if err != nil {
+		log.Error(err, err.Error())
+		r.status.SetDegraded("Failed to get kube controllers gateway secret", err.Error())
+		return reconcile.Result{}, false, err
+	}
+
+	kubeControllerKibanaPublicCertSecret, err := utils.GetSecret(ctx, r.client, render.KibanaPublicCertSecret, rmeta.OperatorNamespace())
+	if err != nil {
+		log.Error(err, err.Error())
+		r.status.SetDegraded("Failed to get Kibana pub cert secret used by kube controllers", err.Error())
+		return reconcile.Result{}, false, err
+	}
+
+	enableESOIDCWorkaround := false
+	if (authentication != nil && authentication.Spec.OIDC != nil && authentication.Spec.OIDC.Type == operatorv1.OIDCTypeTigera) ||
+		esLicenseType == render.ElasticsearchLicenseTypeBasic {
+		enableESOIDCWorkaround = true
+	}
+
+	managerInternalTLSSecret, err := utils.ValidateCertPair(r.client,
+		common.CalicoNamespace,
+		render.ManagerInternalTLSSecretName,
+		render.ManagerInternalSecretKeyName,
+		render.ManagerInternalSecretCertName,
+	)
+
+	if managementCluster != nil {
+		var err error
+		svcDNSNames := dns.GetServiceDNSNames(render.ManagerServiceName, render.ManagerNamespace, r.clusterDomain)
+		svcDNSNames = append(svcDNSNames, render.ManagerServiceIP)
+		certDur := 825 * 24 * time.Hour // 825days*24hours: Create cert with a max expiration that macOS 10.15 will accept
+
+		managerInternalTLSSecret, err = utils.EnsureCertificateSecret(
+			render.ManagerInternalTLSSecretName, managerInternalTLSSecret, render.ManagerInternalSecretKeyName, render.ManagerInternalSecretCertName, certDur, svcDNSNames...,
+		)
+
+		if err != nil {
+			r.status.SetDegraded(fmt.Sprintf("Error ensuring internal manager TLS certificate %q exists and has valid DNS names", render.ManagerInternalTLSSecretName), err.Error())
+			return reconcile.Result{}, false, err
+		}
+	}
+	kubeControllersCfg := kubecontrollers.KubeControllersConfiguration{
+		K8sServiceEp:                 k8sapi.Endpoint,
+		Installation:                 install,
+		ManagementCluster:            managementCluster,
+		ManagementClusterConnection:  managementClusterConnection,
```

Comment:
This component should never be rendered if there's a managementClusterConnection, as that means it's a managed cluster.

If you're only passing this information because the regular kube-controllers sets it, I don't think you should do it. Just don't set it.